pallets/process-validation/src/lib.rs

```diff
@@ -79,53 +90,110 @@ pub mod pallet {
+            let new_version: i32 = Pallet::<T>::_update_version(id.clone()).unwrap();
+            Pallet::<T>::_persist_process(&id, &new_version, restrictions.clone());
 
-            Self::deposit_event(Event::ProcessCreated);
-            Ok(().into())
+            // wrap event data into a struct?
+            Self::deposit_event(Event::ProcessCreated(id, new_version, restrictions, new_version == 1));
+
+            return Ok(().into());
         }
 
-        // TODO: implement disable_process with correct parameters and impl
         #[pallet::weight(T::WeightInfo::disable_process())]
-        pub(super) fn disable_process(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+        pub(super) fn disable_process(
+            origin: OriginFor<T>,
+            id: T::ProcessIdentifier,
+            version: i32,
+        ) -> DispatchResultWithPostInfo {
             T::DisableProcessOrigin::ensure_origin(origin)?;
-            Self::deposit_event(Event::ProcessDisabled);
-            Ok(().into())
+
+            // TODO implement errors
+            // TODO if no process for this version return latest along with the error message
+            if !<ProcessModel<T>>::contains_key(id.clone(), version) {
+                Self::deposit_event(Event::ProcessDisabled(id, version, false));
+                return Ok(().into());
+            }
+
+            let mut updated = false;
+
+            <ProcessModel<T>>::mutate(id.clone(), version, |process| {
+                updated = process.status == ProcessStatus::Disabled;
+                (*process).status = ProcessStatus::Disabled;
+            });
+
+            Self::deposit_event(Event::ProcessDisabled(id, version, updated));
+            return Ok(().into());
         }
     }
+
+    // helper methods
+    impl<T: Config> Pallet<T> {
+        pub fn _get_version(id: T::ProcessIdentifier) -> i32 {
+             return <VersionModel<T>>::get(&id);
+         }
+     
+        pub fn _update_version(id: T::ProcessIdentifier) -> Result<i32, ()> {
+            let version: i32 = Pallet::<T>::_get_version(id.clone());
+            let exists: Result<Process, ()> = <ProcessModel<T>>::try_get(id.clone(), version);
+            // TODO change logic
+            let new_version: i32 = version + if exists.is_ok() { 0 } else { 1 };
+            if new_version == 1 {
+                <VersionModel<T>>::insert(&id, Version { version: new_version });
+            } else {
+                <VersionModel<T>>::mutate(&id, |v| *v = new_version);
+            }
+     
+            return Ok(new_version);
+         }
+         
+         pub fn _persist_process(id: &T::ProcessIdentifier, version: &i32, restrictions: Restrictions) {
+            <ProcessModel<T>>::insert(
+                id,
+                version,
+                Process {
+                    restrictions: restrictions.clone(),
+                    ..Default::default()
+                },
+            );
+        }
+     }
 }
```

Comment:
I like separating this out a lot! One minor comment is on the accessibility and naming. Do these need to be `pub fn` and I'd suggest dropping the leading underscore. In rust leading underscrores are used to make unused variables so I can see this being confusing

Reply:
They do for couple of reasons 
1. allows to use iin pallet otherwise you cannot call `Pallet::_helper_fn` and allows to use in unit testing, thought private should cover but it wasn't will give it another go shortly